queens/drivers/jobscript_driver.py

```diff
@@ -255,16 +298,21 @@ def _run_executable(job_id, execute_cmd, log_file, error_file, verbose=False):
             streaming=verbose,
             raise_error_on_subprocess_failure=False,
         )
+        if self.raise_error_on_jobscript_failure and process_returncode:
+            raise SubprocessError(
+                f"The jobscript with job ID {job_id} has failed with exit code "
+                f"{process_returncode}. The stderr is:\n{stderr}"
+            )
```

Comment:
I like this a lot. I think we should also add stdout to this message. From experience, it turns out that stdout and stderr are not always used consistently, meaning valuable information is lost if not provided in our error message.  Also, I've added the command that was used to make sure you called what you actually wanted.

When I introduced the `SubprocessError`, I added the method `construct_error_from_command`, which does this for you. You can call the method and pass ` f"The jobscript with job ID {job_id} has failed with exit code {process_returncode}` using the `additional_message` argument.